apps/nycbus/nyc_bus.star

```diff
@@ -0,0 +1,159 @@
+"""
+Applet: NYC Bus
+Summary: NYC Bus departures
+Description: Real-time bus departures for your preferred stop.
+Author: samandmoore
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+API_KEY = "<INSERT_API_KEY_HERE>"
+STOP_CODE = "550685"
+BUSTIME_STOP_TIMES_URL = "http://bustime.mta.info/api/siri/stop-monitoring.json"
+BUSTIME_STOP_INFO_URL = "http://bustime.mta.info/api/where/stop/%s.json"
+
+def get_schema():
+    stop_options = [
+        schema.Option(
+            display = "21 ST/BROADWAY SW",
+            value = "550685",
+        ),
+        schema.Option(
+            display = "21 ST/41 AV NE",
+            value = "550670",
+        ),
+    ]
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Location(
+                id = "location",
+                name = "Location",
+                desc = "Location to use for nearby stops.",
+                icon = "place",
+            ),
+            schema.Dropdown(
+                id = "stop_code",
+                name = "Stop Code",
+                desc = "The stop code to use.",
+                icon = "bus",
+                default = stop_options[1].value,
+                options = stop_options,
+            ),
+        ],
+    )
```

Comment:
I filed an issue #110 because I could use some help figuring out how to replicate the behavior of the NYC Subway app. At the moment this is just hardcoding 2 options and pointlessly asking for your location. I'd like to use your location to populate a list of stops to pick from. There's an MTA Bus Times API for that, I just don't know if it's possible to use the location to power the options in the schema if the user needs to provide the location?

Reply:
You're looking for `LocationBased`: https://github.com/tidbyt/pixlet/blob/main/docs/schema/schema.md#locationbased